### PR TITLE
fix kind create cluster failed on linux when do integrate test manually

### PR DESCRIPTION
**Pull request by zackzhangkai**
Synced from https://github.com/istio/istio/pull/35862/files.

Signed-off-by: zackzhangkai <zhangkaiamm@gmail.com>

### CODE REVIEW

File: files/common/Makefile.common.mk [hunk 1]
```diff
@@ -72,7 +72,7 @@ tidy-go:
 	@go mod tidy
 
 mod-download-go:
-	@-GOFLAGS="-mod=readonly" go mod download
+	@GOFLAGS="-mod=readonly" go mod download
```
Comment (howardjohn) [inline files/common/Makefile.common.mk:75]: can you revert or explain why its changed?

File: files/common/scripts/kind_provisioner.sh [hunk 2]
```diff
@@ -37,6 +37,8 @@ DEFAULT_KIND_IMAGE="gcr.io/istio-testing/kindest/node:v1.19.1"
 # COMMON_SCRIPTS contains the directory this file is in.
 COMMON_SCRIPTS=$(dirname "${BASH_SOURCE:-$0}")
 
+KUBECONFIG=${KUBECONFIG:-$HOME/.kube/config}
```
Comment (howardjohn) [inline files/common/scripts/kind_provisioner.sh:40]: Why is this needed? Isn't kind creating a new kubeconfig anyways?

Comment (zackzhangkai) [reply files/common/scripts/kind_provisioner.sh:40]: Kind will create files on `/config/xxx`, which will have no permission. The errors is :

ERROR: failed to create cluster: failed to write KUBECONFIG: open /config/fb379ac4: read-only file system

File: files/common/Makefile.common.mk (hunk 1 above)
Comment (zackzhangkai) [reply files/common/Makefile.common.mk:75]: It just wants to set the GOFLAGS variable, no need `-`.

More details , can run `make gen`  to get.

Comment (howardjohn) [reply files/common/Makefile.common.mk:75]: it does actually: https://stackoverflow.com/questions/28305609/what-does-prefix-mean-in-makefile

Comment (zackzhangkai) [reply files/common/Makefile.common.mk:75]: In the case '@-', if the command got failed, it will continue. But it will occur error on the going. The errors such as "should use GOFLAGS=readonly" or "go mod vendor". 

So maybe no '-' is more suitable.

Comment (ericvn) [reply files/common/Makefile.common.mk:75]: Looking back, the `-` seemed to be on `make lint-license` and not the other targets. In combining the target commands, it seems the `-` was adding to most of the targets. I'm not sure if that was intentional or not.

You might want to verify at least `make license-lint` works on other repos (I know it did on istio/istio when the change was there).

I agree that removing the `-` so the error isn't ignored seems to be the correct thing, unless we know a particular target expects the error and we need the `-` to ignore it.

File: files/common/scripts/kind_provisioner.sh (hunk 2 above)
Comment (ericvn) [reply files/common/scripts/kind_provisioner.sh:40]: I would need to see if this breaks what some are doing when running locally. Need to see if this pollutes the `config` file every time the command is run (ie. new configurations are added). When they get created in the config directory, it is easy to delete them after shutting down the clusters.

Comment (zackzhangkai) [reply files/common/scripts/kind_provisioner.sh:40]: I can reproduce every time. 

do `make shell` to run a virtual environment(docker), then run `./prow/integ-suite-kind.sh` to start Kind clusters.

Comment (ericvn) [reply files/common/scripts/kind_provisioner.sh:40]: @zackzhangkai That's exactly what we do. In our case we run something like ` ARTIFACTS=$PWD/artifacts ./prow/integ-suite-kind.sh --topology MULTICLUSTER --skip-build --skip-cleanup --topology-config prow/config/topology/external-istiod.json`. When complete, I can do a:
```
ls artifacts/
config-config.yaml  config-external.yaml  kind	kubeconfig  topology-config.json  trace.yaml
```
which has the kubeconfig.
I do make a copy of the topology file locally and enter the kubeconfig in that copy and use that on the go tests invocation.
